attestation-agent/deps/crypto/src/rust/ec.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2024 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::{
+    ec::{Curve, KeyWrapAlgorithm},
+    AES_GCM_256_KEY_BITS,
+};
+
+use aes_gcm::aead::generic_array::GenericArray;
+use aes_kw::{Kek, KekAes256};
+use anyhow::{anyhow, Result};
+use p256::{
+    ecdh::diffie_hellman,
+    elliptic_curve::sec1::FromEncodedPoint,
+    pkcs8::{DecodePrivateKey, EncodePrivateKey, LineEnding},
+    EncodedPoint, PublicKey as P256PublicKey, SecretKey,
+};
+use zeroize::Zeroizing;
+
+#[derive(Clone, Debug)]
+pub enum EcKeyPair {
+    P256(P256EcKeyPair),
+}
+
+impl Default for EcKeyPair {
+    fn default() -> Self {
+        Self::P256(P256EcKeyPair::default())
+    }
+}
+
+impl EcKeyPair {
+    pub fn curve(&self) -> Curve {
+        match self {
+            Self::P256(_) => Curve::P256,
+        }
+    }
+
+    pub fn secret_key(&self) -> &SecretKey {
```

Comment:
For openssl implementation this method is called `private_key`. Any reason for this to be different?

Reply:
This is to keep the naming consistent with the underlying library. The Openssl series likes to call the private key secret key, while Rust calls it private key.